packages/admin-ui/src/filters/filter/filter.state.tsx

```diff
@@ -84,11 +99,26 @@ export interface FilterOption<T> {
   value?: T
 }
 
+export type FilterStatus =
```

Comment:
What's the difference between error/error-search and loading/loading-search?

Reply:
Error with search has a different ui, same for loading

Reply:
Maybe there is a UI issue here. That's why its so complex.

Reply:
I don't think the design is unreasonable... The search states "loading" and "error" are different than global loading and global error. The difference between the search states is that the popover keeps showing children (it keeps the searchbox and buttons rendered)

Maybe I implemented it in a confusing way? These are all the states:

```
export type FilterStatus =
  | 'error' // replaces popover with error
  | 'loading' // replaces popover with skeleton, for initial loading of any kind of filter
  | 'ready' 
  | 'empty' // totally empty filter, no options to choose. This replaces the popover with a message
  | 'not-found' // shows not found message, but still shows searchbox and everything else. Doesn't interrupt the user typing
  | 'search-error' // shows error message, but still shows searchbox and everything else
  | 'loading-search' // just shows loading on searchbox
  ```